edweiss-app/components/quiz/QuizComponents.tsx

```diff
@@ -32,53 +32,82 @@ export const MCQDisplay: ReactComponent<{ exercise: Quizzes.MCQ, selectedIds: nu
 
-			<For each={exercise.propositions} key={exercise.question}>
+			<For each={exercise.propositions}>
 				{(proposition, index) =>
-					<CoolCheckBox key={proposition.id} value={selectedIds.includes(index)} onChange={b => {
-						handleSelection(index)
-
-					}} label={<RichText>{proposition.description}</RichText>} />
+					<CoolCheckBox
+						key={proposition.id} value={selectedIds.includes(index)} onChange={b => {
+							handleSelection(index)
+
+						}} label={
+							<TView flexDirection='row' flexColumnGap={12} alignItems='center'>
+								<TText size={"sm"} bold color='subtext0'>
+									{"ABCDEFGHIJKLMNOPQRSTUVWXYZ".charAt(index)}
+								</TText>
+								<TView flex={1}>
+									<RichText>{proposition.description}</RichText>
+								</TView>
+							</TView>
+						} />
 				}
 
 			</For>
 		</TView>
 	);
 });
 
-export const MCQResultDisplay: ReactComponent<{ exercise: Quizzes.MCQ, selectedIds: number[], results: number[]; }> = ({ exercise, selectedIds, results }) => {
+export const MCQResultDisplay: ReactComponent<{ exercise: Quizzes.MCQ, selectedIds: number[], results: number[], disableBottomBar?: boolean }> = ({ exercise, selectedIds, results, disableBottomBar }) => {
 	return (
-		<TView mb={"xs"} bb={1} borderColor='surface0' m={"md"} radius={'lg'} p={"md"}>
 
-			<TView mb={"xs"} p={"md"}>
-				<RichText size={"lg"}>
-					{exercise.question}
-				</RichText>
+		<TView mb={"xs"} bb={disableBottomBar ? 0 : 1} borderColor='surface0' radius={'lg'} pb={"md"} mx={12}>
+
+			<TView mb={12} p={"sm"}>
+				<TText>
+					{`${exercise.question}`}
+				</TText>
+				<TText size={"sm"} color='subtext0' bold>
+					{exercise.numberOfAnswers + " " + t('quiz:quiz_display.answer')}
+				</TText>
 			</TView>
 
-			<For each={exercise.propositions} key={exercise.question}>
+			<For each={exercise.propositions}>
 				{(proposition, index) =>
-					<TView key={exercise.question + proposition.id}
-						backgroundColor={checkResultColor(checkMCQPropositionCorrect(selectedIds, results, index))}
-						mb={"md"} mr={"md"} ml={"md"} p={"sm"} px={"md"}
-						radius={"xl"}>
-						<TText key={exercise.question + proposition.id} color={textColor(checkResultColor(checkMCQPropositionCorrect(selectedIds, results, index)))}>
-							{proposition.description}
-						</TText>
-					</TView>}
+					<CoolCheckBox
+						borderWidth={2}
+						key={proposition.id} value={selectedIds.includes(index)} disabled
+						activeColor={checkResultColor(checkMCQPropositionCorrect(selectedIds, results, index))}
+						iconName={checkMCQPropositionCorrect(selectedIds, results, index) == "wrong" ? "close" : "checkmark"}
+						label={
+							<TView flexDirection='row' flexColumnGap={12} alignItems='center'>
+								<TText size={"sm"} bold color='subtext0'>
+									{"ABCDEFGHIJKLMNOPQRSTUVWXYZ".charAt(index)}
+								</TText>
+								<TView flex={-1}>
+									<RichText>{proposition.description}</RichText>
+								</TView>
+							</TView>
```

Comment:
This code is almost the same as the one in lines 56-63. You maybe can merge it in one rendering function taking the desired flex as parameter e.g.

`const coolCheckBoxLabel(flex: number) { ... }`